src/main/java/com/truelayer/java/payments/entities/paymentdetail/AttemptFailedPaymentDetail.java

```diff
@@ -0,0 +1,44 @@
+package com.truelayer.java.payments.entities.paymentdetail;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.truelayer.java.entities.AuthorizationFlowWithConfiguration;
+import com.truelayer.java.entities.PaymentSource;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AttemptFailedPaymentDetail extends PaymentDetail {
+
+    Status status = Status.ATTEMPT_FAILED;
+
+    PaymentSource paymentSource;
+
+    ZonedDateTime failedAt;
+
+    FailureStage failureStage;
+
+    String failureReason;
+
+    AuthorizationFlowWithConfiguration authorizationFlow;
+
+    @JsonGetter
+    public Optional<AuthorizationFlowWithConfiguration> getAuthorizationFlow() {
+        return Optional.ofNullable(authorizationFlow);
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    public enum FailureStage {
```

Comment:
With respect to `Failed`, this new status can be reached only from `authorizing` and `authorized` (see [here](https://docs.truelayer.com/docs/web-sdk#payment-retries-for-web-sdk))